src/main/scala/io/hydrosphere/mist/master/FrontendJobExecutor.scala

```diff
@@ -52,6 +53,10 @@ class FrontendJobExecutor(
   val common: Receive = {
     case GetActiveJobs =>
       sender() ! jobs.values.map(i => JobExecutionStatus(i.request.id, name, status = i.status))
+    case FailRemainingJobs =>
+      jobs.foreach { case (id, info) =>
+        statusService ! StatusMessages.FailedEvent(id, System.currentTimeMillis(), "worker down (e.g. timeout, jvm crash)")
+      }
```

Comment:
There are additional things that we need to do to complete job inside `FrontendJobExecutor` -  see `onJobDone` method.